msr/msre/multiphysics_core_model/steady_state/neu.i

```diff
@@ -0,0 +1,654 @@
+    execute_on = 'transfer timestep_end'
+  []
+  [absorption_RR]
+    type = FluxRxnIntegral
+    block = 'core lower_plenum upper_plenum down_comer riser pump elbow'
+    cross_section = sigma_absorption
+    coupled_flux_groups = ' sflux_g0    sflux_g1    sflux_g2    sflux_g3 sflux_g4    sflux_g5    sflux_g6    sflux_g7 sflux_g8    sflux_g9    sflux_g10   sflux_g11 sflux_g12   sflux_g13   sflux_g14   sflux_g15'
+    execute_on = 'transfer timestep_end'
+  []
+  [fission_RR]
+    type = FluxRxnIntegral
+    block = 'core lower_plenum upper_plenum down_comer riser pump elbow'
+    cross_section = sigma_fission
+    coupled_flux_groups = ' sflux_g0    sflux_g1    sflux_g2    sflux_g3 sflux_g4    sflux_g5    sflux_g6    sflux_g7 sflux_g8    sflux_g9    sflux_g10   sflux_g11 sflux_g12   sflux_g13   sflux_g14   sflux_g15'
+    execute_on = 'transfer timestep_end'
+  []
+  [ngamma_RR]
+    type = FluxRxnIntegral
+    block = 'core lower_plenum upper_plenum down_comer riser pump elbow'
+    cross_section = sigma_ngamma
+    coupled_flux_groups = ' sflux_g0    sflux_g1    sflux_g2    sflux_g3 sflux_g4    sflux_g5    sflux_g6    sflux_g7 sflux_g8    sflux_g9    sflux_g10   sflux_g11 sflux_g12   sflux_g13   sflux_g14   sflux_g15'
+    execute_on = 'transfer timestep_end'
+  []
+  [Leakage]
+    type = PartialSurfaceCurrent
+    boundary = 'right bottom loop_boundary pump_outlet downcomer_inlet top_core_barrel'
+    transport_system = transport
+  []
+[]
+
+# ==============================================================================
+# EXECUTION PARAMETERS
+# ==============================================================================
+
+[Preconditioning]
+  [SMP]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  type = Eigenvalue
+  solve_type = PJFNKMO
+
+  petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart'
+  petsc_options_value = 'hypre boomeramg 50'
+  l_max_its = 5000
+  nl_abs_tol = 1e-10
+
+  free_power_iterations = 4 # important to obtain fundamental mode eigenvalue
+
+  # MultiApp
+  fixed_point_min_its = 2
+  fixed_point_max_its = 50
+  fixed_point_rel_tol = 1e-6
+  fixed_point_abs_tol = 1e-6
+
+[]
+
+# ==============================================================================
+# MULTIAPPS AND TRANSFERS
+# ==============================================================================
+[MultiApps]
+  [flow_dnp]
+    type = FullSolveMultiApp
+    input_files = 'th.i'
+    execute_on = 'timestep_end'
+    max_procs_per_app = 48
+    keep_solution_during_restore = true
+  []
+[]
+
+[Transfers]
+  [power_density]
+    type = MultiAppShapeEvaluationTransfer
+    to_multi_app = flow_dnp
+    source_variable = power_density
+    variable = power_density
+    execute_on = 'timestep_end'
+  []
+  [fission_source]
+    type = MultiAppShapeEvaluationTransfer
+    to_multi_app = flow_dnp
+    source_variable = fission_source
+    variable = fission_source
+    execute_on = 'timestep_end'
+  []
+  [c1]
```

Comment:
I m pretty sure we can now transfer directly from the TH app c_i to an array variable c_i, dropping like 100 lines from this input